specs/protocol/isthmus/exec-engine.md

```diff
@@ -131,6 +132,16 @@ is applicable for scenarios where the actual withdrawals root value is not readi
 
 Withdrawals list in the block body is encoded as an empty RLP list.
 
+## EVM Changes
+
+Similar to the `bn256Pairing` precompile in the [granite hardfork](../granite/exec-engine.md),
+[EIP-2537](https://eips.ethereum.org/EIPS/eip-2537) introduces a BLS
+precompile that short-circuits depending on input size in the EVM.
+
+The `BLS12-381` curve operation precompile reverts if its input is
+larger than `235008` bytes. This is the input size that consumes
+approximately `20M` gas given the latest `BLS12-381` gas schedule on L2.
```

Comment:
Is this the only curve operation we're accelerating? G1_MSM/G2_MSM have variable input length and could cause a large increase in `op-program` runtime if a ton of multiplications are passed in, but I guess the 20M gas limit only matters if they're accelerated precompiles. What's the downside to accelerating all the BLS precompiles?

Reply:
Good question - I think other operations could be accelerated. CC @Inphi @ajsutton would appreciate your input here

Reply:
We should benchmark the others in the proof for sure, but pairing checks tend to be really expensive in the prover even with the miller loop optimization, so that'd be the first one I'm worried about.

But for example, bn128 add enables the worst-case block construction (at least in the samples that @BrianBland ran in his benchmarks), so we should also look into the others.

Reply:
Discussed with @ajsutton and @clabby today - we cannot support these precompiles in kona for `no_std` without a significant amount of work, so we've largely settled on accelerating all these precompiles.

I'd say as such the priority of the benchmarks can be downgraded to nice-to-have. If your low on bandwidth, it's not critical.